Update lisflood example notebook

SarahAlidoost opened the pull request.

SarahAlidoost committed:
polish the notebook

SarahAlidoost committed:
fix links, re-run the notebook

SarahAlidoost committed:
revise minor things in text

SarahAlidoost commented:
@sverhoeven or @Peter9192 who has time to review this PR? thanks.

sverhoeven commented on docs/examples/lisflood.ipynb:
```diff
@@ -5,131 +5,178 @@
    "id": "c02a5bfe",
```
<ol><li>noetbook -&gt; notebook</li><li>italic 1990 -&gt; surround file name with backticks</li><li>Why did you make a copy of forcing files why not use them directly, similar to the parameters?</li><li>Can we describe how these forcing files where made for example: </li></ol><p> </p><p>Forcing files on Cartesius where made by running ESMValTool recipe and lisvap in the past . We can now use those files to run the Lisflood model.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/eWaterCycle/ewatercycle/pull/175/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

SarahAlidoost replied:
<ol><li>it is fixed. </li><li>it is fixed.</li><li>we don't have lisvap files in comparison directory and different directories in forcing.load results in error in model.initialize later.</li><li>it is added.</li></ol>

SarahAlidoost replied:
3.  So, in ewatercycle, something like this:

```python
forcing = ewatercycle.forcing.load_foreign(target_model='lisflood',
                                           directory='/projects/0/wtrcycle/comparison/lisflood/forcing/',
                                           start_time='1990-01-01T00:00:00Z',
                                           end_time='1990-12-31T00:00:00Z',
                                           forcing_info={
                                               'PrefixPrecipitation': 'lisflood_ERA-Interim_pr_1990_1990.nc',
                                               'PrefixTavg': 'lisflood_ERA-Interim_tas_1990_1990.nc',
                                               'PrefixE0': '/scratch/shared/ewatercycle/lisflood_example/lisflood_forcing_data/lisflood_e0_1990_1990.nc',
                                               'PrefixES0': '/scratch/shared/ewatercycle/lisflood_example/lisflood_forcing_data/lisflood_es0_1990_1990.nc',
                                               'PrefixET0': '/scratch/shared/ewatercycle/lisflood_example/lisflood_forcing_data/lisflood_et0_1990_1990.nc',
                                           }
                                          )
```

does not work. In comparison, we could have different directories for esmvaltool output and lisvap output files. I think because we had used the absolute paths in the config file before.

sverhoeven replied:
ah, yep, lisvap output files will not be mounted inside containers. OK, then a copy is what we need until #96 is completed.

sverhoeven commented on docs/examples/lisflood.ipynb:
```diff
@@ -5,131 +5,178 @@
    "id": "c02a5bfe",
```
<p>You are zooming in on some catchment Can you mention in the top cell which catchment we ran lisflood for?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/eWaterCycle/ewatercycle/pull/175/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

SarahAlidoost replied:
<p>An explanation is added. </p>

sverhoeven commented on docs/examples/lisflood.ipynb:
```diff
@@ -5,131 +5,178 @@
    "id": "c02a5bfe",
```
<p>Doing a spatial selection might warrant a description in markdown</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/eWaterCycle/ewatercycle/pull/175/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

SarahAlidoost replied:
<p>The description is added.</p>

sverhoeven commented on docs/examples/lisflood.ipynb:
```diff
@@ -5,131 +5,178 @@
    "id": "c02a5bfe",
```
<p>Can you explain where target_longitude&nbsp;=&nbsp;[-71.35]</p><p>target_latitude&nbsp;=&nbsp;[42.64] is? Merrimack observation station</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/eWaterCycle/ewatercycle/pull/175/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

SarahAlidoost replied:
<p>the explanation is added.</p>

sverhoeven reviewed (changes requested):
Notebook looks nice, I like the in time and spatial plots. Notebook ran successfully on Cartesius.

The while loop has some extra output `22 (estimated simulation ...`, which I expect is from the model in combination with the new Jupyter version. When I ran it I only got isostrs due older Jupyter version.

The config file was written with the setup arguments.

See reviewnb comments for suggestions.

SarahAlidoost committed:
revise or add text in markdown cells

SarahAlidoost commented:
> Notebook looks nice, I like the in time and spatial plots. Notebook ran successfully on Cartesius.
> 
> The while loop has some extra output `22 (estimated simulation ...`, which I expect is from the model in combination with the new Jupyter version. When I ran it I only got isostrs due older Jupyter version.
> 
> The config file was written with the setup arguments.
> 
> See reviewnb comments for suggestions.

Thank you for reviewing. I addressed your comments, please have another look.

sverhoeven reviewed (approved):
Looks good to me.

Except replace `model.__del__()` with `model.finailize()`.

SarahAlidoost commented:
> Looks good to me.
> 
> Except replace `model.__del__()` with `model.finailize()`.

Thanks. `model.finailize()` calls `.bmi.finalize()` which does not work for lisflood, please see [this issue](https://github.com/eWaterCycle/era5-comparison/issues/153).
